src/main/kotlin/ecommerce/service/MemberService.kt

```diff
@@ -30,9 +32,11 @@ class MemberService(
         val hashedPassword = passwordService.hashPassword(request.password)
         val member = request.toModel(hashedPassword)
 
-        val savedCart = Cart(member = member)
-        member.assignCart(savedCart)
         val savedMember = memberRepository.save(member)
+        val savedCart = cartRepository.save(Cart(member = savedMember))
+        savedMember.assignCart(savedCart)
+        memberRepository.save(savedMember)
```

Comment:
If you’ve set up `Cascade`, try using it; if not, consider removing the `Cascade`. 🙂